src/lib/core/partialcubeop.test.ts

```diff
@@ -3,145 +3,145 @@ import * as partial from "./partialcubeop";
-  },
-  center: {
-    face: 0,
-    row: 1,
-    column: 1,
-  },
-  right: {
-    face: 0,
-    row: 1,
-    column: 2,
-  },
-  bottomLeft: {
-    face: 0,
-    row: 2,
-    column: 0,
-  },
-  bottom: {
-    face: 0,
-    row: 2,
-    column: 1,
-  },
-  bottomRight: {
-    face: 0,
-    row: 2,
-    column: 2,
+  positions: {
+    topLeft: {
+      face: 0,
+      row: 0,
+      column: 0,
+    },
+    top: {
+      face: 0,
+      row: 0,
+      column: 1,
+    },
+    topRight: {
+      face: 0,
+      row: 0,
+      column: 2,
+    },
+    left: {
+      face: 0,
+      row: 1,
+      column: 0,
+    },
+    center: {
+      face: 0,
+      row: 1,
+      column: 1,
+    },
+    right: {
+      face: 0,
+      row: 1,
+      column: 2,
+    },
+    bottomLeft: {
+      face: 0,
+      row: 2,
+      column: 0,
+    },
+    bottom: {
+      face: 0,
+      row: 2,
+      column: 1,
+    },
+    bottomRight: {
+      face: 0,
+      row: 2,
+      column: 2,
+    },
   },
 };
 
 function expectAllOtherFacesUnaffected(
   cubeFace: typeof CUBE_FACE_3X3 | typeof CUBE_FACE_2X2,
   operation: (cubeIndex: CubeIndex, face: Face, dimension: number) => CubeIndex,
 ): void {
-  Object.entries(cubeFace)
-    .filter(([key, _]) => key !== "dimension")
-    // For each CubeIndex on the given cube face...
-    .forEach(([key, value]) => {
-      const cubeIndex = value as CubeIndex;
-
-      // For each face other than this one...
-      FACES.filter(face => face !== cubeIndex.face).forEach(face => {
-        // Compute the corresponding CubeIndex on that face...
-        const copyOnOtherFace = {
-          face,
-          row: cubeIndex.row,
-          column: cubeIndex.column,
-        };
-
-        // ...and expect it to remain in place when this face is turned
-        it(`leaves ${key} of face ${face} in place`, () => {
-          expect(
-            operation(copyOnOtherFace, cubeIndex.face, CUBE_FACE_3X3.dimension),
-          ).toEqual(copyOnOtherFace);
-        });
+  // For each position on the given cube face...
+  Object.entries(cubeFace.positions).forEach(([name, cubeIndex]) => {
+    // For each face other than this one...
+    FACES.filter(face => face !== cubeIndex.face).forEach(face => {
+      // Compute the corresponding CubeIndex on that face...
+      const copyOnOtherFace = {
+        face,
+        row: cubeIndex.row,
+        column: cubeIndex.column,
+      };
+
+      // ...and expect it to remain in place when this face is turned.
+      it(`leaves ${name} of face ${face} in place`, () => {
+        expect(
+          operation(copyOnOtherFace, cubeIndex.face, cubeFace.dimension),
+        ).toEqual(copyOnOtherFace);
       });
     });
+  });
 }
 
 describe("2x2 clockwise quarter turn", () => {
   it("sends top left corner to top right corner", () => {
     expect(
       partial.clockwiseQuarterTurn(
-        CUBE_FACE_2X2.topLeft,
-        CUBE_FACE_2X2.topLeft.face,
+        CUBE_FACE_2X2.positions.topLeft,
```

Comment:
not necessary, but you can pull out positions in the beginning of the describe block so you don't have to key in so deep every time.

```ts
describe("2x2 clockwise quarter turn", () => {
	const positions = CUBE_FACE_2X2.positions;

	it("sends top left corner to top right corner", () => { ... }
}
```

Reply:
Also done.